dotnet/src/Extensions/PromptTemplates.Liquid/LiquidPromptTemplate.cs

```diff
@@ -25,13 +33,34 @@ public LiquidPromptTemplate(PromptTemplateConfig config)
         this._config = config;
     }
 
+    /// <inheritdoc/>
     public Task<string> RenderAsync(Kernel kernel, KernelArguments? arguments = null, CancellationToken cancellationToken = default)
     {
         Verify.NotNull(kernel);
 
         var template = this._config.Template;
         var liquidTemplate = Template.ParseLiquid(template);
-        var nonEmptyArguments = arguments?.Where(x => x.Value is not null).ToDictionary(x => x.Key, x => x.Value!);
+        Dictionary<string, object> nonEmptyArguments = new();
+        foreach (var p in this._config.InputVariables)
```

Comment:
nit: Since `RenderAsync` method is big enough, I would probably put this logic in separate method, that will return `nonEmptyArguments`. Same is related to blocks below. If some block of code requires comments, it's usually a sign to put it in separate method with good naming and XML doc if needed.

Reply:
I'm going to merge so I can test the functionality, verify tests, etc. I'll take a look at making this change in a follow up PR